src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -434,6 +443,15 @@ private static string GenerateUpgradeReport(NuGetProject nuGetProject, string ba
 
                     uiService.EndOperation();
 
+                    // don't show "Succeeded" if we actually cancelled...
+                    if ((!continueAfterPreview) || (!acceptedLicense))
```

Comment:
Feels like this will read better if the conditions were inverted.

Reply:
For what it's worth, I don't agree. I think this is a more direct implementation of "if either message box was cancelled....". `(!(continueAfterPreview && acceptedLicense))` feels weird to me :)